packages/plugins/plugin-deck/src/components/DeckLayout/DeckLayout.tsx

```diff
@@ -81,9 +83,32 @@ export const DeckLayout = ({
     }
   }, [layoutParts]);
 
-  const searchEnabled = !!usePlugin('dxos.org/plugin/search');
-  const activeId = useMemo(() => Array.from(attention.attended ?? [])[0], [attention.attended]);
+  const [activeId, setActiveId] = useState<string | undefined>(Array.from(attention.attended ?? [])[0]);
+  const deckRef = useRef<HTMLDivElement>(null);
+  useLayoutEffect(() => {
+    if (layoutMode === 'deck' && activeId) {
+      const el = deckRef.current?.querySelector(`article[data-attendable-id="${activeId}"]`);
+      // TODO(burdon): Generalize scroll into view by parent.
+      setTimeout(() => {
+        el?.scrollIntoView({ behavior: 'smooth', inline: 'center' });
+      }, 100); // TODO(burdon): Hack. The element already exists but scrolling doesn't happen without the delay. Calculate or memo offset?
+    }
+  }, [layoutMode, activeId]);
```

Comment:
A few things here:

It feels funky having a default value from attention but then getting subsequent values from the scroll into view event. What's the intention with always supplying a value?

Since we're in deck land here, you can watch the `scrollIntoView` property from the layout context instead. It's in scope here, just not destructured.

Make sure to return a clean up function that clears the timeout. It's a shame we need to wait so long.

Reply:
let's discuss how to do this properly.

Reply:
can you take this. i think i have things working again, but i would be good to remove the need for this timeout altogether.